benchflow-faban-client/src/main/java/cloud/benchflow/faban/client/responses/RunStatus.java

```diff
@@ -42,9 +45,13 @@ public RunStatus(String statusCode, RunId runId) {
       case "DENIED":
         this.status = StatusCode.DENIED;
         break;
+      case "UNKNOWN":
+        this.status = StatusCode.UNKNOWN;
+        break;
       default:
         throw new IllegalRunStatusException(
-            "RunId " + runId + "returned illegal run status " + statusCode);
+            "RunId " + runId + "returned illegal run status " + formattedStatusCode,
```

Comment:
The point is that `UNKNOWN` is a possible status of the client. I rework this also according to https://github.com/benchflow/benchflow/pull/442#pullrequestreview-45544110.